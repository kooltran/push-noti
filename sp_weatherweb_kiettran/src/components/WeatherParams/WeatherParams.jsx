import React from "react";

const WeatherParams = () => {
  return (
    <div className="weatherParams-wrapper">
      <div className="weatherParams-item">
        <div className="title">PSI</div>
        <div className="content bg-numb">23</div>
        <div className="extra">Good</div>
      </div>
      <div className="weatherParams-item">
        <div className="title">RAIN</div>
        <div className="content normal-numb">0</div>
        <div className="extra">mm</div>
      </div>
      <div className="weatherParams-item">
        <div className="title">DENGUE</div>
        <div className="content circle-shape"></div>
        <div className="extra"></div>
      </div>
      <div className="weatherParams-item">
        <div className="add-icon"></div>
        <div className="add-text">Add</div>
      </div>
    </div>
  );
};

export default WeatherParams;
