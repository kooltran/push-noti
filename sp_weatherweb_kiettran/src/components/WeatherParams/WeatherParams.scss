.weatherParams {
  &-wrapper {
    display: flex;
  }

  &-item {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    height: 100px;
    padding: 10px 0;
    border-right: 1px solid $COLOR_WHITE;

    .title {
      font-weight: 500;
      font-size: 18px;
    }

    .bg-numb {
      font-size: 24px;
      padding: 8px;
      background: $COLOR_GREEN;
      border-radius: 4px;
      width: 30px;
      height: 30px;
    }

    .normal-numb {
      font-size: 30px;
    }

    .circle-shape {
      width: 40px;
      height: 40px;
      border: 2px solid $COLOR_WHITE;
      border-radius: 100%;
      background-color: $COLOR_LIGHT_BLUE;
    }

    .add-icon {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 30px;
      height: 30px;
      border: 1px solid $COLOR_BLACK;
      border-radius: 100%;
      text-align: center;
      cursor: pointer;

      &:before {
        display: inline-block;
        width: 20px;
        height: 20px;
        background-image: url($ICON_PLUS);
        background-size: contain;
        content: "";
      }
    }

    .add-text {
      color: $COLOR_BLACK;
      font-weight: 500;
      font-size: 20px;
    }

    &:last-child {
      border-right: 0;
      justify-content: space-around;
    }
  }
}
