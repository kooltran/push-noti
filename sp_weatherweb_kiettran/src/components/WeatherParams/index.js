import WeatherParams from "./WeatherParams";

export default WeatherParams;
