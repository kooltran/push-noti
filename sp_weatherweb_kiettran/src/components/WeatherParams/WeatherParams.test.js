import React from "react";
import renderer from "react-test-renderer";
import WeatherParams from "./WeatherParams";

it("should render WeatherParams correctly", () => {
  const tree = renderer.create(<WeatherParams />).toJSON();
  expect(tree).toMatchSnapshot();
});
