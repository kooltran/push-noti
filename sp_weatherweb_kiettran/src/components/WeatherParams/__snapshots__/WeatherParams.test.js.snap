// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WeatherParams correctly 1`] = `
<div
  className="weatherParams-wrapper"
>
  <div
    className="weatherParams-item"
  >
    <div
      className="title"
    >
      PSI
    </div>
    <div
      className="content bg-numb"
    >
      23
    </div>
    <div
      className="extra"
    >
      Good
    </div>
  </div>
  <div
    className="weatherParams-item"
  >
    <div
      className="title"
    >
      RAIN
    </div>
    <div
      className="content normal-numb"
    >
      0
    </div>
    <div
      className="extra"
    >
      mm
    </div>
  </div>
  <div
    className="weatherParams-item"
  >
    <div
      className="title"
    >
      DENGUE
    </div>
    <div
      className="content circle-shape"
    />
    <div
      className="extra"
    />
  </div>
  <div
    className="weatherParams-item"
  >
    <div
      className="add-icon"
    />
    <div
      className="add-text"
    >
      Add
    </div>
  </div>
</div>
`;
